tests/e2e/vm_creation_performance_test.go

```diff
@@ -0,0 +1,228 @@
+	GinkgoHelper()
+	resVMD, err := client.VirtualMachineDisks(namespace).Create(context.TODO(), &vmd, metav1.CreateOptions{})
+	Expect(err).NotTo(HaveOccurred(), "cannot create vmd - %s", &vmd.Name)
+	resVM, err := client.VirtualMachines(namespace).Create(context.TODO(), &vm, metav1.CreateOptions{})
+	Expect(err).NotTo(HaveOccurred(), "cannot create vm - %s", &vm.Name)
+	fmt.Println("VM", resVM.Name, "with VMD", resVMD.Name, "created")
+}
+
+var _ = Describe("Performance test 20 vm creation", Label("performance"), Ordered, ContinueOnFailure, func() {
+	const (
+		vmName                         = "perf-test-vm"
+		vmdName                        = "perf-test-vmd"
+		cvmiName                       = "ubuntu-22.04"
+		vmCount                        = 26
+		overallTimeout                 = 12 * time.Minute
+		deleteGracePeriodSeconds int64 = 30
+	)
+
+	var cvmi *v1alpha2.ClusterVirtualMachineImage
+	var vmMap = make(map[string]string)
+
+	clientConfig := kubeclient.DefaultClientConfig(&pflag.FlagSet{})
+	client, err := kubeclient.GetClientFromClientConfig(clientConfig)
+	if err != nil {
+		log.Fatalf("Cannot obtain Virtualization client: %v\n", err)
+	}
+
+	AfterAll(func() {
+		By("Delete all resources")
+		vmList, err := client.VirtualMachines(conf.Namespace).List(context.TODO(), metav1.ListOptions{})
+		for _, vm := range vmList.Items {
+			err = client.VirtualMachines(conf.Namespace).Delete(context.TODO(), vm.Name, metav1.DeleteOptions{
+				GracePeriodSeconds: ptr.To(deleteGracePeriodSeconds)})
+			Expect(err).NotTo(HaveOccurred())
+			if err != nil {
+				fmt.Println(err)
+			}
+		}
+	})
+
+	Context("VM", func() {
+		cvmi, err = CVMI(client, cvmiName, "create")
+		Expect(err).NotTo(HaveOccurred(), "%s", err)
+
+		It("Create", func() {
+			for i := 1; i <= vmCount; i++ {
+				vmd := VMD(conf.Namespace, fmt.Sprintf("%s-%d", vmdName, i), *cvmi)
+				vm := VM(conf.Namespace, fmt.Sprintf("%s-%d", vmName, i), vmd)
+				createVM(client, vm, vmd, conf.Namespace)
+			}
+		})
+		It("Wait until all virtual machines are running", func() {
+			vmList, err := client.VirtualMachines(conf.Namespace).List(context.TODO(), metav1.ListOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			start := time.Now()
+			fmt.Println("Starting at", start)
+
+			for len(vmMap) != vmCount {
+				vmList, err = client.VirtualMachines(conf.Namespace).List(context.TODO(), metav1.ListOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				now := time.Now()
+				elapsed := now.Sub(start)
+
+				for _, vm := range vmList.Items {
+					if vmMap[vm.Name] != vm.Name && string(vm.Status.Phase) == "Running" {
+						vmMap[vm.Name] = string(vm.Status.Phase)
+					}
+				}
+
+				if elapsed > overallTimeout {
+					break
+				}
+
+				time.Sleep(1 * time.Second)
+			}
+			fmt.Println("Finished at", time.Now())
+			fmt.Println("Duration", time.Now().Sub(start))
+			Expect(len(vmList.Items)).To(Equal(vmCount))
```

Comment:
It can be simpler with Eventually that has overall timeout and an error checking.

https://onsi.github.io/ginkgo/#patterns-for-asynchronous-testing
https://onsi.github.io/gomega/#eventually

Quick draft:
```
Eventually(func() (int, error) {
  vmList, err := client.VirtualMachines(conf.Namespace).List(context.TODO(), metav1.ListOptions{})
  if err != nil {
    return 0, err
  }

  runningCount := 0
  for _, vm := range vmList.Items {
    if string(vm.Status.Phase) == "Running" {
      runningCount++
    }
  }
  return runningCount, nil
}).WithTimeout(overallTimeout).Should(Equal(vmCount))
```

Overall duration better to calculate in AfterAll block: it will report duration even for failed test.